client.go

```diff
@@ -0,0 +1,109 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	ec "github.com/bitcoin-sv/go-sdk/primitives/ec"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/auth"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/httpx"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+)
+
+// Client provides methods for user-related and admin-related APIs.
+// This struct is designed to abstract and simplify the process of making HTTP calls
+// to the relevant endpoints. By utilizing this Client struct, developers can easily
+// interact with both user and admin APIs without needing to manage the details
+// of the HTTP requests and responses directly.
+type Client struct {
+	userAPI *user.API
+}
+
+// SharedConfig retrieves the shared configuration from the user configurations API.
+// This method constructs an HTTP GET request to the "/shared" endpoint and expects
+// a response that can be unmarshaled into the response.SharedConfig struct.
+// If the request fails or the response cannot be decoded, an error will be returned.
+func (c *Client) SharedConfig(ctx context.Context) (*response.SharedConfig, error) {
+	res, err := c.userAPI.ConfigsAPI.SharedConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve shared configuration from User Configs API: %w", err)
+	}
+	return res, nil
+}
+
+// NewWithXPub creates a new client instance using an extended public key (xPub).
+// Generates a hierarchical deterministic (HD) key from the provided xPub and constructs
+// the necessary configuration for the API instance. The SignRequest flag is set to false,
+// indicating that requests made with this instance will not be signed.
+func NewWithXPub(addr, xPub string) (*Client, error) {
+	key, err := bip32.GetHDKeyFromExtendedPublicKey(xPub)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate HD key from xPub: %w", err)
+	}
```

Comment:
Total minor, but I like one line break after if statement -> for me it's a little bit cleaner especially if this "if statement" contains:
break/continue/return.

Reply:
@wregulski  Do you mean something like this? 


```
func (c *Client) SharedConfig(ctx context.Context) (*response.SharedConfig, error) {
	res, err := c.userAPI.ConfigsAPI.SharedConfig(ctx)
	if err != nil {
		return nil, fmt.Errorf("failed to retrieve shared configuration from User Configs API: %w", err)
	}
	
	return res, nil
}
```